helpers/getInferredName.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+var getInferredName;
+try {
+	// eslint-disable-next-line no-new-func
+	getInferredName = Function('s', 'return { [s]() {} }[s].name;');
```

Comment:
Why does this use the Function constructor on strings instead of just a function expression?

Reply:
because that would be a syntax error if the engine didn't support the syntax; that's what the try/catch is for.